### PR TITLE
feat(Styleguide): Docs Navigation - GM-133

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview

This adds a bunch of navigation features to our storybook, including:
- Table of Contents on every index page with enriched meta information (components and status).
- Breadcrumbs at every page level, to make it easier to figure out what subjection you are in at a glance.
- Adds a `subtitle` parameter to each story which is shown at the top of the document and in the sections card in the table of contents.

Stories:
- Each component has a filled out `subtitle` string as a short description.  This can be any valid markdown and formatted correctly.
- Edited some of the story hierarchy to improve structure.

### Usage

Adding a table of contents is as simple as dropping docs block in any page:
```tsx
import { TableOfContents } from '~styleguide/blocks';

<TableOfContents />
```

The component will automatically find all relevant stories/sections to the current context and display their subsections with some added content.

### Screenshots

**Breadcrumbs**
localhost_6006__path=_docs_gamut--page (1)

**Top Level ToC**
localhost_6006__path=_docs_gamut--page

* Adds `TableOfContents` Doc Block to our styleguide storybook components.
* Adds breadcrumb to each Docs page.
* Adds a subtitle parameter to Story parameters so descriptions are available.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: GM-133
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Testing

1. Visit https://5ff8d1826b57e70fd9e52bcc--gamut-preview.netlify.app
2. Click through some of the content sections.
3. And navigate back up using the `Breadcrumbs` at the top of the page.
4. Ensure things make sense.

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff33f416c7a414994546738--gamut-preview.netlify.app>https://5ff33f416c7a414994546738--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff33f416c7a414994546738>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.4bc8bd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff3633577220971828becc8--gamut-preview.netlify.app>https://5ff3633577220971828becc8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff3633577220971828becc8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.064f5d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff4aa848cb8434c7ab5befb--gamut-preview.netlify.app>https://5ff4aa848cb8434c7ab5befb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff4aa848cb8434c7ab5befb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff4c4dfdd13fc00f41c1822--gamut-preview.netlify.app>https://5ff4c4dfdd13fc00f41c1822--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff4c4dfdd13fc00f41c1822>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.1f8893.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff4c6d7a75f2a008e84c8d4--gamut-preview.netlify.app>https://5ff4c6d7a75f2a008e84c8d4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff4c6d7a75f2a008e84c8d4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.3.8-alpha.84a3a6.0<br/>@codecademy/gamut-labs@7.6.4-alpha.84a3a6.0<br/>@codecademy/gamut@19.8.1-alpha.84a3a6.0<br/>@codecademy/styleguide@21.7.1-alpha.84a3a6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff4e89ec6c16e21eac74883--gamut-preview.netlify.app>https://5ff4e89ec6c16e21eac74883--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff4e89ec6c16e21eac74883>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.3.8-alpha.e9042d.0<br/>@codecademy/gamut-labs@7.6.4-alpha.e9042d.0<br/>@codecademy/gamut@19.8.1-alpha.e9042d.0<br/>@codecademy/styleguide@21.7.1-alpha.e9042d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff4f746a75f2a325b84d16b--gamut-preview.netlify.app>https://5ff4f746a75f2a325b84d16b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff4f746a75f2a325b84d16b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.a399c8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff5181dcc383410baf5b723--gamut-preview.netlify.app>https://5ff5181dcc383410baf5b723--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff5181dcc383410baf5b723>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.e49c70.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff5223017fc1222c567349d--gamut-preview.netlify.app>https://5ff5223017fc1222c567349d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff5223017fc1222c567349d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.ec17c5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff52699cc383430baf5b226--gamut-preview.netlify.app>https://5ff52699cc383430baf5b226--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff52699cc383430baf5b226>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.b1f1cd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff5d30619b6e5150bce8e27--gamut-preview.netlify.app>https://5ff5d30619b6e5150bce8e27--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff5d30619b6e5150bce8e27>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.a447d3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff72bef94ef651c436b5bae--gamut-preview.netlify.app>https://5ff72bef94ef651c436b5bae--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff72bef94ef651c436b5bae>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.e8de63.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff8a361b03b467580f4dcd4--gamut-preview.netlify.app>https://5ff8a361b03b467580f4dcd4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff8a361b03b467580f4dcd4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.434e8c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff8a5511f76742c3bdd2a30--gamut-preview.netlify.app>https://5ff8a5511f76742c3bdd2a30--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff8a5511f76742c3bdd2a30>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.c72506.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff8c198917aae012146f116--gamut-preview.netlify.app>https://5ff8c198917aae012146f116--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff8c198917aae012146f116>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.ca48fc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff8c8fd23574c00b1ec6764--gamut-preview.netlify.app>https://5ff8c8fd23574c00b1ec6764--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff8c8fd23574c00b1ec6764>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.7.1-alpha.ddfe40.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ff8d1826b57e70fd9e52bcc--gamut-preview.netlify.app>https://5ff8d1826b57e70fd9e52bcc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ff8d1826b57e70fd9e52bcc>Deploy Logs</a></p>

File: packages/styleguide/.storybook/components/Page/index.tsx [hunk 1]
```diff
@@ -35,41 +38,88 @@ const HeaderRow = styled.div`
...
+
+    const nextLink = {
+      text: kind,
+      kind: `${prevKind}${kind}`,
+    };
+
+    links.push(nextLink);
+  });
+
+  links = [{ text: 'Gamut', kind: 'Gamut' }, ...links];
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/components/Page/index.tsx:65]: ⛳ ✨  _(I haven't checked if this works exactly, but generally something like this?)_

```suggestion
  const links = [{ text: 'Gamut', kind: 'Gamut' }];

  path.forEach((kind, i) => {
    const prevKind = i ? `${links[i].kind}/` : '';

    const nextLink = {
      text: kind,
      kind: `${prevKind}${kind}`,
    };

    links.push(nextLink);
  });
```

**Review by JoshuaKGoldberg — APPROVED** [review]
This is dope. I'm a little saddened that there are no emojis anymore 😛 , but can live with adding them as a followup. (We'd talked about using visual regression tests as image sources)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.8.3-alpha.a158ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ffc7e3f20fc872f9381d076--gamut-preview.netlify.app>https://5ffc7e3f20fc872f9381d076--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5ffc7e3f20fc872f9381d076>Deploy Logs</a></p>

**Pull request merged**